src/app/flashcard/components/pages/create-flashcard-page/create-flashcard-page.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Observable, of, Subscriber, Subscription } from 'rxjs';
+import { Topic } from 'src/app/flashcard/model/topic';
+import { TopicService } from 'src/app/flashcard/service/topic.service';
+import { FlashcardService } from '../../../service/flashcard.service';
+
+@Component({
+  selector: 'app-create-flashcard-page',
+  templateUrl: './create-flashcard-page.component.html',
+  styleUrls: ['./create-flashcard-page.component.css']
+})
+export class CreateFlashcardPageComponent implements OnInit {
+
+  topics: Topic[] = [];
+  subscription: Subscription;
+  anonymous = false;
+
+  form: any = {
+    userId: null,
+    topicID: {},
+    question: null,
+    difficulty: null
+  };
+
+  constructor(private flashcardService: FlashcardService, private topicService: TopicService) {
+    this.subscription = new Observable<any>().subscribe();
+    this.topicService.getAll().subscribe(res => this.topics = res);
+  }
+
+
+  ngOnInit(): void {
+  }
+
+  onSubmit(): void {
+
+    // TODO: set creator to current user
+    this.form.userId = 1;
+
+    switch (this.form.difficulty) {
+      case 'easy':
+        this.form.difficulty = 1;
+        break;
+      case 'medium':
+        this.form.difficulty = 2;
+        break;
+      case 'hard':
+        this.form.difficulty = 3;
+        break;
+    }
+
+    console.log(this.form);
+
+    this.subscription = this.flashcardService.create(this.form).subscribe(
+      data => {
+        console.log(data);
```

Comment:
After creation, this should route back to the Flashcard page. Add flashcard router as another parameter in the constructor. 
this.flashcardrouter.navigate(['/flashcard'])